### PR TITLE
Feature/router error handling

### LINKED ISSUE #11: As a user, I can click the forward and back arrows to navigate between pages.
As a user,  when I have clicked on a movie I can use the back button in my browser to navigate back to the main page and then can use the forward button in my browser to navigate back to the movie that I had last clicked. 

## Acceptance Criteria: 
-[] As a user, while on a movieDisplay page, when I click on the back arrow in my browser I should see the main page with all the movies displayed.
-[] As a user, after navigating back to main from movieDisplay, when I click the forward button in my arrow in my browser I should see the movie I last clicked on displayed (on the movieDisplay page)

### PR DESCRIPTION
**Pull request by stephaniemagdic**
# Thank you for pushing your code!
## What (if any) features are you implementing?
- Install react-router-dom in order to refactor our conditional rendering to render based on url.
- Now there are three `pages` that are rendered based on what url the user is at.

This closes #11 , #10 , #9 , #3, #6 , #7 

## What (if anything) did you refactor?
- We converted our movieDisplay component to be a class component in order to hold state for the selected movie and included a fetch request based on the url in componentDidMount.

## Were there any issues that arose?
- N/A

## Is there anything that you need from your teammate?
- N/A

## Any other comments, questions, or concerns?
- We want to make our error display page more dynamic by passing in props to tailor feedback.
- We are considering creating an Express microserver which will mean we need to refactor how our components load data on page load.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
- [x] open index.html
- [x] console.log()
- [x] dev tools

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: cypress/integration/dashboard_spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+describe('Dashboard User Flows', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3001');
+  });
+
+  it('Should display movie poster and ratings on individual movie cards on page load', () => {
+    cy.intercept('GET', 'https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
```
Comment (MTCawthray) [inline cypress/integration/dashboard_spec.js:7]: Nice intercept here. We discussed if a page load is technically a 'user flow'. I think at very least it is important that we did this in order to solidify the concept in our heads!

File: cypress/integration/dashboard_spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      statusCode: 201,
+      body: {
+         movies: [ 
+          {
+            average_rating: 6.142857142857143,
+            backdrop_path: "https://image.tmdb.org/t/p/original//pq0JSpwyT2URytdFG0euztQPAyR.jpg",
+            id: 694919,
+            poster_path: "https://upload.wikimedia.org/wikipedia/commons/6/6e/Golde33443.jpg",
```
Comment (MTCawthray) [inline cypress/integration/dashboard_spec.js:15]: cute puppy pic :)

File: cypress/integration/dashboard_spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      }
+    })
+    cy.request('http://localhost:3001')
+    .get('article').eq(0).contains('Rating: 6.14')
+    cy.get('article').eq(1).contains('Rating: 5.27')
+  })
+
+  it('User should be able to click on a movie which will take them away from the page', () => {
+    cy.get('main').contains('RANCID TOMATILLOS')
+    cy.get('section').get('article').eq(0).click().url().should('include', '/694919')
```
Comment (MTCawthray) [inline cypress/integration/dashboard_spec.js:37]: I wonder if we should do this same test to the second card in the array just to have full coverage? We can always come back to this on the refactor later this weekend.

File: cypress/integration/movieDetails_spec.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    .get('article').contains('Money Plane')
+    .get('article').contains("A professional thief with $40 million in debt and his family's life on the line must commit one final heist - rob a futuristic airborne casino filled with the world's most dangerous criminals.")
+  })
+
+  it('User should be able to click the back button icon and be taken back to the main page', () => {
+    cy.visit('http://localhost:3001/movies/694919')
+    cy.get('.backButton').click().url().should('not.include', 'movies')
+  })
+
+  it('If user alters a url path to an undefined path, it should bring them to an error display page', () => {
```
Comment (MTCawthray) [inline cypress/integration/movieDetails_spec.js:36]: Glad we were able to test this sad path. If/when we get to our iteration 5 place of displaying a tailored error message, we will be sure to come back and test for that scenario.

File: cypress/integration/dashboard_spec.js (hunk 3 above)
Comment (stephaniemagdic) [reply cypress/integration/dashboard_spec.js:37]: I think that's a great idea.

File: cypress/integration/dashboard_spec.js (hunk 2 above)
Comment (stephaniemagdic) [reply cypress/integration/dashboard_spec.js:15]: We might want to add a second puppy pic to our second card to make sure the data is somewhat different than our actual data.

File: src/App/App.js [hunk 5]
```diff
@@ -1,81 +1,47 @@
...
-     selectedMovie: null
+     error: null
    }
  }
 
  componentDidMount = () => {
   fetchData('movies').then(data => {
     this.setState({movies: [...this.state.movies, ...data.movies]})
   }).catch((data) => this.setState({error: data.message}))
-  // The above .catch is setting our state, however the react dev tools simply shows it as an empty object. We should revisit this after we make our error display component.
```
Comment (MTCawthray) [inline src/App/App.js:25]: It is amazing how much less code is needed now that we've refactored for Router!

File: src/Header/Header.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ const Header = () => {
   return (
     <header>
       <h1>
-        ROTTEN TOMATILLOS
+        RANCID TOMATILLOS
```
Comment (MTCawthray) [inline src/Header/Header.js:7]: Glad we got the name of the project right 🤣 My bad!!!

File: src/MovieDetail/MovieDetail.css [hunk 7]
```diff
@@ -0,0 +1,17 @@
+.movieDetail {
```
Comment (MTCawthray) [inline src/MovieDetail/MovieDetail.css:1]: This still needs some fine tuning and what not, but a great proof of concept for the time being.

File: src/MovieDisplay/MovieDisplay.js [hunk 8]
```diff
@@ -1,23 +1,59 @@
...
 import MovieDetail from '../MovieDetail/MovieDetail';
+import Header from '../Header/Header';
+import { fetchData } from '../util.js';
+import { Component } from 'react';
+import  ErrorDisplay from '../ErrorDisplay/ErrorDisplay';
+import { Link } from 'react-router-dom';
 
 
-const MovieDisplay = (props) => {
+class MovieDisplay extends Component {
```
Comment (MTCawthray) [inline src/MovieDisplay/MovieDisplay.js:10]: Converting this to a class component was the right move in our situation. When we learn Hooks, we should revisit this project and refactor to use those instead!

**Review by MTCawthray — APPROVED** [review]
Great work as always, Stephanie! I've left a few comments, but all in all I think we are good to merge and move on. I'm excited to get rolling on iteration 5!

**Pull request merged**